app/views/play/level/LevelChatView.coffee

```diff
@@ -262,11 +262,23 @@ module.exports = class LevelChatView extends CocoView
       @reallySaveChatMessage({ text, sender })
       @savingChatMessage = undefined
 
+  cleanUpApiProperties: (chat) ->
+    context = chat.context
+    currentCode = Object.values(context.code.current)[0]
+    solutionCode = Object.values(context.code.solution)[0]
```

Comment:
How does this `context.code.current` and `context.code.solution` objects look like?
I didn't try this, but are we sure `Object.values` and not `Object.keys` is the right function to call here? 
Also we're getting the `[0]`th element. Are  we sure it'll be an array, and not a string - so we can call the `.include` function later?

Reply:
They are objects of programming language to code, like `{python: "# some code"}`. Taking the first one should usually work, since the names of the functions don't really change between levels.

However, `solutionCode` might not exist, so we should guard against that. (Summit's Gate is a good test level, since there is no solution, and players have many APIs on their gear by that point.) If there is no solution code, we probably want to either:
1. Not get rid of any APIs, if there aren't very many APIs;
2. Get rid of all APIs that aren't in the current code, if there are very many APIs such that the prompt wouldn't fit; or
3. Not include any API documentation, if the current code is short/empty and there are very many APIs, and maybe adapt our AI prompt to not pretend like it has included the APIs.

Alternatively, we could switch to a compressed API format when there are too many, like just giving the name instead of the full API doc for each method/property.

Reply:
good catch. will update soon